### PR TITLE
Require authentication when user post deals

### PR DESCRIPTION
**Pull request by limli**
Check if user is logged in when posting deal. If user is not logged in, respond with SC_UNAUTHORIZED. Else if user is logged in, use the user's userId as posterId for the posted deal.

### CODE REVIEW

**Review by jyx11011 — DISMISSED** [review]
LGTM

File: src/main/webapp/WEB-INF/jsp/add-deal.jsp [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <meta name="description" content="" />
   <meta name="author" content="" />
   <title>Add Deal</title>
+  <script src="/js/util.js"></script>
```
Comment (SuyashLakhotia) [inline src/main/webapp/WEB-INF/jsp/add-deal.jsp]: Should this change be in a separate PR?

File: src/test/java/com/google/step/servlets/DealPostServletTest.java [hunk 2]
```diff
@@ -50,34 +56,47 @@
   private HttpServletRequest request;
   private DealPostServlet servlet;
   private DealManager dealManager;
+  private UserService userService;
+  private UserManager userManager;
+  private HttpServletResponse response;
+  private PrintWriter writer;
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/DealPostServletTest.java]: It's good practice to prefix mocked classes with `mock` (e.g. `mockRequest`).

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/jsp/add-deal.jsp (hunk 1 above)
Comment (limli) [reply src/main/webapp/WEB-INF/jsp/add-deal.jsp]: the util.js file contains the authentication for the frontend side. It doesn't need to be in this PR, but I thought it was somewhat relevant.

**Review by SuyashLakhotia — APPROVED** [review]
